openmdao/core/tests/test_check_derivs.py

```diff
@@ -2287,6 +2287,68 @@ def test_cs(self):
         self.assertTrue('9.7743e+00' in compact_lines[11],
             "'9.7743e+00' not found in '%s'" % compact_lines[11])
 
+    def test_check_totals_show_progress(self):
+        prob = om.Problem()
+        prob.model = SellarDerivatives()
+        prob.model.nonlinear_solver = om.NonlinearBlockGS()
+
+        prob.model.add_design_var('x', lower=-100, upper=100)
+        prob.model.add_design_var('z', lower=-100, upper=100)
+        prob.model.add_objective('obj')
+        prob.model.add_constraint('con1', upper=0.0)
+        prob.model.add_constraint('con2', upper=0.0)
+
+        prob.set_solver_print(level=0)
+
+        prob.setup(force_alloc_complex=True)
+
+        prob.model.nonlinear_solver.options['atol'] = 1e-15
+        prob.model.nonlinear_solver.options['rtol'] = 1e-15
+
+        # We don't call run_driver() here because we don't
+        # actually want the optimizer to run
+        prob.run_model()
+
+        # check derivatives with complex step and a larger step size.
+        stream = StringIO()
+        totals = prob.check_totals(method='fd', show_progress=True, out_stream=stream)
+
+        lines = stream.getvalue().splitlines()
+        self.assertTrue(lines[0], "1/3: Checking derivatives with respect to: '_auto_ivc.v0 [0]' ... 0.0088 seconds")
+        self.assertTrue(lines[1], "2/3: Checking derivatives with respect to: '_auto_ivc.v0 [1]' ... 9.2891 seconds")
```

Comment:
A few things:

1. Don't check the time -- that can change.  To make a check like that work, you can do `assertTrue(small_string in large_string)`
2. You should convert the '_auto_ivc.v0' name into its corresponding connected promoted input name. Otherwise this has no meaning.
3. For the "false" check, just assert that the text "Checking derivatives with respect to" isn't found anywhere in the output.  You can glue the stream output together into one string with ` "\n".join(lines)` and do a single check